Install missing `conventional-changelog-conventionalcommits` preset and update CI for `semantic-release`

**Description:**  
Our GitHub Actions build is failing during the **generateNotes** step of `@semantic-release/release-notes-generator` with:

```
Error: Cannot find module 'conventional-changelog-conventionalcommits'
    at importFrom …
    at async default (…/load-changelog-config.js:27:63)
    …
  code: 'MODULE_NOT_FOUND',
  pluginName: '@semantic-release/release-notes-generator'
```

We need to ensure the required preset is installed and that our CI installs devDependencies.

---

**Step-by-step instructions:**

1. **Install the missing preset**  
   ```bash
   # From your repo root
   npm install --save-dev conventional-changelog-conventionalcommits
   # or with Yarn
   yarn add -D conventional-changelog-conventionalcommits
   ```

2. **Verify your `semantic-release` config**  
   - Open your release configuration (e.g. `.releaserc.json` or the `"release"` block in `package.json`).  
   - Ensure the `release-notes-generator` plugin references the correct preset:
     ```jsonc
     {
       "plugins": [
         "@semantic-release/commit-analyzer",
         [
           "@semantic-release/release-notes-generator",
           { "preset": "conventionalcommits" }
         ],
         "@semantic-release/npm",
         "@semantic-release/github"
       ]
     }
     ```

3. **Update your CI workflow**  
   - In your GitHub Actions (or other CI), replace any `npm ci --only=production` step with a full install, for example:
     ```yaml
     - name: Install dependencies
       run: npm ci              # pulls in devDependencies too
     ```
   - Commit and push the updated workflow file.

4. **Run & verify**  
   - Trigger a new CI run.  
   - Confirm the `generateNotes` step completes without a “MODULE_NOT_FOUND” error.

---

**Definition of Done:**

- [ ] `conventional-changelog-conventionalcommits` is listed under `devDependencies`  
- [ ] Release config references `"preset": "conventionalcommits"` with correct spelling  
- [ ] CI workflow installs all dependencies (including devDependencies)  
- [ ] CI build passes the `generateNotes` step without errors  
- [ ] (Optional) Update any docs or README about the release process

--- a/package.json
+++ b/package.json
@@ -23,6 +23,7 @@
     "@semantic-release/github": "^11.0.1",
     "@semantic-release/release-notes-generator": "^14.0.3",
     "concurrently": "^8.2.0",
+    "conventional-changelog-conventionalcommits": "^8.0.0",
     "semantic-release": "^24.2.3"
   }
 }

